core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryDateHeaderItemViewHolder.kt

```diff
@@ -14,10 +14,10 @@ class HistoryDateHeaderItemViewHolder(itemView: View) : BaseViewHolder<DateItem>
     val yesterdayDate = todaysDate.minusDays(1)
     val givenDate = LocalDate.parse(item.dateString, DateTimeFormatter.ofPattern("d MMM yyyy"))
 
-    when {
-      todaysDate == givenDate -> header_date.setText(R.string.time_today)
-      yesterdayDate == givenDate -> header_date.setText(R.string.time_yesterday)
-      else -> header_date.text = item.dateString
+    when (givenDate) {
+      todaysDate -> header_date.setText(R.string.time_today)
+      yesterdayDate -> header_date.setText(R.string.time_yesterday)
+      else -> header_date.setText(item.dateString)
```

Comment:
this should complain at you for not using property access syntax, prefer property access syntax when it is available.

Reply:
It did, I changed it to match the set texts above to keep it consistent. Can change back. Also, should I squish my commits? Think I have made a lot of tiny ones.

Reply:
by making it consistent you have made it more difficult to understand, it makes it look like they are all doing the same thing if you just scan the method names. If I wasn't paying attention I would probably ask you to rewrite it so that the `when` would return the `stringId` which would be incorrect because the `else` branch sets the text with a string.

While all execution branches "set the text of a textview" they do it differently and the original code expressed that better.

Squash your commits if you want to, I don't mandate it on this project as I prefer the history to not be pretty lies

Reply:
Ok, that is a valid point. Will change and not squish! Thanks again for your feedback on all this, I value it greatly.

Reply:
Happy to be of use, code is hard and nobody teaches quality.